src/server/game/Entities/Pet/Pet.cpp

```diff
@@ -1520,7 +1520,7 @@ bool Pet::removeSpell(uint32 spell_id, bool learn_prev, bool clear_ab)
     }
 
     // if remove last rank or non-ranked then update action bar at server and client if need
-    if (m_charmInfo->RemoveSpellFromActionBar(spell_id) && !learn_prev && clear_ab)
```

Comment:
I changed it back to what is was. intentionally. order matters in short-circuit boolean evaluation and executing  RemoveSpellFromActionBar() in any case broke the pet action bar autoccast settings.